+hw/DaqController.m

```diff
@@ -48,53 +51,75 @@
     SampleRate = 1000 % output sample rate ("scans/sec") of the daq device
         % 1000 is also the default of the ni daq devices themselves, so if
         % you don't change this, it doesn't actually do anything. 
+        
   end
   
   properties (Transient)
+    
     DaqSession % should be a DAQ session containing at least one analogue output channel
     DigitalDaqSession % a DAQ session containing only digital output channels
+    ClockDaqSession % a DAQ session for implementing output to a clock channel
+    
   end
   
   properties (Dependent)
+    
     Value %The current voltage on each DAQ channel
     NumChannels %Number of channels controlled
     AnalogueChannelsIdx %Logical array of analogue channel IDs
+    
   end
   
   properties (Access = private, Transient)
+    
     CurrValue
+    
   end
   
   methods
+    
     function createDaqChannels(obj)
-      if isempty(obj.DaqSession)
+      if isempty(obj.DaqSession)&&any(strncmp('ao',(obj.DaqChannelIds),2))
         obj.DaqSession = daq.createSession('ni');
         obj.DaqSession.Rate = obj.SampleRate; 
       end
-      if isempty(obj.DigitalDaqSession)&&any(~obj.AnalogueChannelsIdx)
+      if isempty(obj.DigitalDaqSession)&&any(strncmp('port',(obj.DaqChannelIds),4))
         obj.DigitalDaqSession = daq.createSession('ni');
       end
+      if isempty(obj.ClockDaqSession)&&any(strncmp('ctr',(obj.DaqChannelIds),3))
+        obj.ClockDaqSession = daq.createSession('ni');
+      end
       n = obj.NumChannels;
       if n > 0
-        for ii = 1:n
+        for i = 1:n
           if iscell(obj.DaqIds)
-            daqid = obj.DaqIds{ii};
+            daqid = obj.DaqIds{i};
           else
             daqid = obj.DaqIds;
           end
-          if obj.AnalogueChannelsIdx(ii) % is channal analogue?
+          % is channel analogue?
+          if strncmp('ao',(obj.DaqChannelIds{i}),2)
             obj.DaqSession.addAnalogOutputChannel(...
-              daqid, obj.DaqChannelIds{ii}, 'Voltage');
-          else % assume digital, always output only
+              daqid, obj.DaqChannelIds{i}, 'Voltage');
+          % is channel clock output?
+          elseif strncmp('ctr',(obj.DaqChannelIds{i}),3)
+            obj.ClockDaqSession.addCounterOutputChannel(...
+              daqid, obj.DaqChannelIds{i}, 'PulseGeneration');
+          else % assume digital, always 'OutputOnly'
             obj.DigitalDaqSession.addDigitalChannel(...
-              daqid, obj.DaqChannelIds{ii}, 'OutputOnly');
+              daqid, obj.DaqChannelIds{i}, 'OutputOnly');
           end
         end
+        
+        % what are these lines doing? why outputSingleScan? why a/d separate?
```

Comment:
It's important to output the default value for instance if you wanted a channel that was always high and occasionally went low, or an analogue channel with a baseline voltage.  Why were lines 94-97 removed?

Reply:
b/c I have it in later (line 121)